ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiCommandService.scala

```diff
@@ -97,7 +100,10 @@ private[apiserver] final class ApiCommandService private (
       ledgerConfig: LedgerConfiguration,
   )(implicit loggingContext: LoggingContext): Future[Completion] = {
     val appId = request.getCommands.applicationId
-    val submitter = TrackerMap.Key(application = appId, party = request.getCommands.party)
+    // Note: command completions are returned as long as at least one of the original submitters
+    // is specified in the command completion request.
+    val party = CommandsValidator.effectiveSubmitters(request.getCommands).actAs.head
+    val submitter = TrackerMap.Key(application = appId, party = party)
```

Comment:
What's preventing us from supporting multiple parties in the tracker already?

Reply:
Nothing. Initially I thought it's easier to not change the command tracker code, and instead if a set of parties submits a command (which should be rare) just reuse the command tracker of one of those parties.

I have added multi-party command trackers in https://github.com/digital-asset/daml/pull/8125/commits/f478634e0fd5e20b0e3e74c1a3f7abb02098ee73